### PR TITLE
Dynamic RNN without padding

### PR DESCRIPTION
**Pull request by ArnoldLIULJ**
### Checklist
- [x] I've tested that my changes are compatible with the latest version of Tensorflow.
- [x] I've read the [Contribution Guidelines](https://github.com/tensorlayer/tensorlayer/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### CODE REVIEW

**Comment by JingqingZ** [comment]
TODO: example needed for Dynamic RNN in the documentation

**Comment by zsdonghao** [comment]
do you mean, we can input a batch of text without using padding?  or we provide a sequence length into the RNN?

File: tensorlayer/layers/recurrent.py [hunk 1]
```diff
@@ -157,49 +158,104 @@ def build(self, inputs_shape):
             self._trainable_weights.append(var)
 
     # @tf.function
-    def forward(self, inputs, initial_state=None, **kwargs):
+    def forward(self, inputs, actual_length=None, initial_state=None, **kwargs):
```
Comment (zsdonghao) [inline tensorlayer/layers/recurrent.py]: use `sequence_length` ?

**Comment by ArnoldLIULJ** [comment]
# change to initial_length
# check length of the argument initial_length

**Comment by zsdonghao** [comment]
can this one work ? https://tensorlayer.readthedocs.io/en/latest/modules/layers.html#compute-sequence-length-2

**Pull request merged**